.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -244,7 +247,11 @@ jobs:
             sudo apt-get install -yqq libllvm14 || true;
         fi
         sudo -E bash devops/scripts/install_drivers.sh llvm/devops/dependencies.json ${{ inputs.install_dev_igc_driver == 'true' && 'llvm/devops/dependencies-igc-dev.json --use-dev-igc' || '' }} --all
+    - name: Setup oneAPI env
+      if: ${{ inputs.is_icx == 'true' }}
+      uses: ./devops/actions/setup_linux_oneapi_env
```

Comment:
Why do we need that in `*run-tests.yml` at all? IMO, it should be enough to have that in build tasks only.

Reply:
because when we build with icx, some of the libraries used added to the path as part of the icx setup are used so if we don't do this we get undefined reference errors in every test

Reply:
andrei gave me a flag that seems to fix this, testing it now. if this works, will be huge

Reply:
fixed in https://github.com/intel/llvm/pull/16652/commits/ac6b279a9220443a3113aab9355cac604f2fee81 thx